### PR TITLE
docs(spin/auth): Adds oauth2 config docs.

### PR DESCRIPTION
**Pull request by jtk54**

### CODE REVIEW

File: guides/spin/cli/index.md [hunk 1]
```diff
@@ -71,4 +71,29 @@ and the [README](https://github.com/spinnaker/spin/blob/master/README.md) for mo
...
+    scopes: # Scopes requested for the token
+    - scope1
+    - scope2
+```
+
+See https://www.spinnaker.io/setup/security/authentication/oauth/providers/ 
+to see examples for acquiring a clientId/clientSecret from your provider.
+
+Unlike X.509, OAuth2 needs initialized once to authenticate with the provider before
+it can be used for automation. To authenticate, configure OAuth2 as shown and execute
```
Comment (lwander) [inline guides/spin/cli/index.md]: "as shown above"

File: guides/spin/cli/index.md [hunk 2]
```diff
@@ -71,4 +71,29 @@ and the [README](https://github.com/spinnaker/spin/blob/master/README.md) for mo
...
+any `spin` command. You will be prompted to authenticate with your OAuth2 provider
+and paste an access code in the terminal the `spin` command is executing in. `spin`
+then exchanges the code for an OAuth2 token, which it caches for future use. All
+subsequent `spin` calls will use the cached OAuth2 token for authentication with
```
Comment (lwander) [inline guides/spin/cli/index.md]: What happens when your token expires?

Comment (lwander) [reply guides/spin/cli/index.md]: Can you share how once this is done, copying the `~/.spin/config` file to another server allows it be used for w/o intervention by some automated system?

File: guides/spin/cli/index.md [hunk 3]
```diff
@@ -71,4 +71,30 @@ and the [README](https://github.com/spinnaker/spin/blob/master/README.md) for mo
...
+    clientSecret: # OAuth2 client secret
+    scopes: # Scopes requested for the token
+    - scope1
+    - scope2
+```
+
+See https://www.spinnaker.io/setup/security/authentication/oauth/providers/ 
+to see examples for acquiring a clientId/clientSecret from your provider.
+
+Unlike X.509, OAuth2 needs initialized once to authenticate with the provider before
```
Comment (dorbin) [inline guides/spin/cli/index.md]: "...needs to be initialized..."

File: guides/spin/cli/index.md [hunk 4]
```diff
@@ -71,4 +71,30 @@ and the [README](https://github.com/spinnaker/spin/blob/master/README.md) for mo
...
+it can be used for automation. To authenticate, configure OAuth2 as shown above and execute
+any `spin` command. You will be prompted to authenticate with your OAuth2 provider
+and paste an access code in the terminal the `spin` command is executing in. `spin`
```
Comment (dorbin) [inline guides/spin/cli/index.md]: I think you can delete "in the terminal the `spin` command is executed in"

**Review by dorbin — APPROVED** [review]
LGTM, after fix the one typo and, optionally, delete the possibly superflous phrase.

**Pull request merged**